ui/hooks/useOriginMetadata.js

```diff
@@ -19,24 +19,38 @@ import { SUBJECT_TYPES } from '../../shared/constants/app';
  * current origin
  */
 export function useOriginMetadata(origin) {
-  const subjectMetadata = useSelector(getSubjectMetadata);
+  const targetSubjectMetadata = useSelector((state) =>
+    getTargetSubjectMetadata(state, origin),
+  );
+
   if (!origin) {
     return null;
   }
 
-  const url = new URL(origin);
-  const minimumOriginMetadata = {
-    host: url.host,
-    hostname: url.hostname,
-    origin,
-    subjectType: SUBJECT_TYPES.UNKNOWN,
-  };
+  let minimumOriginMetadata;
+  try {
+    const url = new URL(origin);
+    minimumOriginMetadata = {
+      host: url.host,
+      hostname: url.hostname,
+      origin,
+    };
+  } catch (_) {
+    // do nothing
+  }
 
-  if (subjectMetadata?.[origin]) {
+  if (targetSubjectMetadata && minimumOriginMetadata) {
+    return {
+      ...minimumOriginMetadata,
+      ...targetSubjectMetadata,
+    };
+  } else if (targetSubjectMetadata) {
+    return targetSubjectMetadata;
+  } else if (minimumOriginMetadata) {
     return {
       ...minimumOriginMetadata,
-      ...subjectMetadata[origin],
+      subjectType: SUBJECT_TYPES.UNKNOWN,
     };
   }
-  return minimumOriginMetadata;
+  return null;
```

Comment:
Well. I don't feel great about this. I wonder if there's an alternative to using this hook for things that aren't actually origins :thinking:

Reply:
I think we need to take some time to consider how to deal with non-URL origins generally, maybe by moving to some generic notion of "subject identifiers" and let `origin` refer to URLs only and be optionally present.

Returning `null` here is more for the sake of completion however, because we should never end up in a situation where an "origin" doesn't have any metadata. Meanwhile, we could throw an error instead?